string name
uint32 nr
